Proton/Sources/Swift/Attachment/Attachment.swift

```diff
@@ -344,14 +345,14 @@ open class Attachment: NSTextAttachment, BoundsObserving {
         var size: CGSize
 
         if let boundsProviding = contentView as? DynamicBoundsProviding {
-            size = boundsProviding.sizeFor(containerSize: adjustedContainerSize, lineRect: adjustedLineFrag)
+            size = boundsProviding.sizeFor(attachment: self, containerSize: adjustedContainerSize, lineRect: adjustedLineFrag)
         } else {
             size = contentView?.bounds.integral.size ?? view.bounds.integral.size
-        }
 
-        if (size.width == 0 || size.height == 0),
-           let fittingSize = contentView?.systemLayoutSizeFitting(adjustedContainerSize) {
-            size = fittingSize
+            if (size.width == 0 || size.height == 0),
```

Comment:
Makes more sense under `else` though I assume some users might find changes in behaviour when `DynamicBoundsProviding` gives an empty size for whatever reason. However, I am sure it should be fixed on a caller, `DynamicBoundsProviding` implementation, side.